### PR TITLE
LPS-82061

### PR DESCRIPTION
**Pull request by wanderlast**
Hi Roberto!

LPP: https://issues.liferay.com/browse/LPP-30355
LPS: https://issues.liferay.com/browse/LPS-82061

This is the pull request that you asked me to send in the SME task. It contains the fix we spoke about (fixing the propositional logic from || to &&). It also applies this logic to the configuration section's role selector since all other configuration sections are currently gated by user visibility besides roles.

Please let me know if you have any questions. Thanks!

### CODE REVIEW

**Comment by wanderlast** [comment]
ci:reopen

**Comment by wanderlast** [comment]
ci:retest

**Comment by robertoDiaz** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/announcements/announcements-web/src/main/resources/META-INF/resources/announcements/configuration.jsp [hunk 1]
```diff
@@ -21,7 +21,7 @@ String tabs1 = ParamUtil.getString(request, "tabs1", "sites");
 
 List<Group> groups = GroupLocalServiceUtil.getUserGroups(user.getUserId(), true);
 List<Organization> organizations = OrganizationLocalServiceUtil.getUserOrganizations(user.getUserId());
-List<Role> roles = RoleLocalServiceUtil.getRoles(PortalUtil.getCompanyId(renderRequest));
```
Comment (robertoDiaz) [inline modules/apps/announcements/announcements-web/src/main/resources/META-INF/resources/announcements/configuration.jsp]: Why do we have to filter the roles this way and not the organizations or the groups, for example??

Does it make sense to restrict the portlet configuration based on the user that is configuring it?? I don't think so.

Comment (wanderlast) [reply modules/apps/announcements/announcements-web/src/main/resources/META-INF/resources/announcements/configuration.jsp]: Groups/organizations/user groups are currently already filtered by user. Roles is the only option that isn't filtered by userId currently in the configuration menu and instead displays all role options. Do you believe it'd be better to change groups/organizations/user groups to display all configuration options instead?

Comment (robertoDiaz) [reply modules/apps/announcements/announcements-web/src/main/resources/META-INF/resources/announcements/configuration.jsp]: I've check the history of this logic and it seems that it was introduced in SOS-1586

Saying this, I think that we should not change this behavior in a fix.

Please, remove this commit and fwd it to @sergiogonzalez 

If you have any concern you could open a new PTR or a PSM if you feel it as a perceived bug

Thanks!!

**Comment by robertoDiaz** [comment]
Please resend (to @sergiogonzalez ) removing the last commit.

Thanks!

**Comment by wanderlast** [comment]
Understood. Thanks for looking at this!

**Comment by wanderlast** [comment]
Resent without last commit: https://github.com/sergiogonzalez/liferay-portal/pull/4491

**Pull request closed without merging**